### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Greg0109**
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button.
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/addremove.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import localStorageConst from './localstorage';
+
+const addremove = {
+  add: (title) => {
```
Comment (Davidosky007) [inline src/addremove.js:4]: * Kindly fix this functionality ` the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).`. currently, when i add a new task it takes the index of the array eg(0,1,2,3) use the array length instead as stated in the project requirement.

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Greg0109, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost git

## Required Changes ♻️
## _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Greg0109** [comment]
the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5) - Done

File: src/index.js [hunk 2]
```diff
@@ -12,13 +13,35 @@ function addEventsDragAndDrop(el) {
...
+    addButton.addEventListener('click', () => {
+      const title = document.getElementById('title').value;
+      addremove.edit(title, index);
+      editUI(false);
+    });
+  } else {
+    const formDiv = document.getElementById('editForm');
+    formDiv.innerHTML = '';
+  }
+}
```
Comment (Dipeshtwis) [inline src/index.js:34]: [OPTIONAL]

- I would recommend to reset the form after adding a toDO list. Kindly the form stays the same.

Screenshot from 2021-07-15 18-54-25

**Review by Dipeshtwis — APPROVED** [review]
# Hi @Greg0109, 

### Status: Approved ✔️

:+1:  Professional repo, with good commit message and PR title, keep up the good work :muscle:
:+1: You have implemented the previous reviewer suggestion.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- I would recommend to add the `language` section as well in README file so that user would know by which language you've built the project.

**_Also check the comments under the review_**


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**